### PR TITLE
Merge the same TOML file writing logic

### PR DESCRIPTION
**Pull request by ESWZY**
## Summary

Seeing that there is the same TOML file writing logic here, for maintainability considerations, they can be merged together.

## Output

No change.

## Documentation

- Should this change be documented?
    - [x] No

### CODE REVIEW

File: internal/build/container_ops.go
Comment (joe-kimmel-vmw) [inline internal/build/container_ops.go:245]: imo the loss of specific error messages here and above is undesirable, but we could fix this by passing in one additional parameter to `writeToml` something like `dataDescriptor string` and it could then be given the value `stack metadata` on line 253 below,  and appropriate values elsewhere.   What do you think?

File: internal/build/container_ops.go
Comment (joe-kimmel-vmw) [inline internal/build/container_ops.go:237]: I think rather than introduce a new public API function only for the tests, it would be better to test one of the existing public functions

**Review by joe-kimmel-vmw — COMMENTED** [review]
Thanks for contributing @ESWZY ! 

It looks like your intention is to "DRY" up this part of the code by merging redundant codepaths?   I think that's a fine idea overall though I'll defer to the actual pack maintainers' judgement.   

I've made a couple of suggestions.  I think as a project we really do want our error messages to be as helpful and informative as possible,  so maintaining the existing specificity of the error messages is probably desired.

**Comment by ESWZY** [comment]
> 

Thanks for your generous suggestions, I will consider it!

**Review by joe-kimmel-vmw — APPROVED** [review]
LGTM  but it'll have to wait for a pack maintainer review :)

**Review by jkutner — APPROVED** [review]

**Pull request merged**
